R/vpc-simulation.R

```diff
@@ -0,0 +1,253 @@
+
+#' Create a simulation model object from an existing model
+#'
+#' @param .mod a `bbi_nonmem_model` object.
+#' @param n number of replicates.
+#' @param seed a seed for simulation.
+#' @param replicate_var a variable name for storing each iteration. Cannot match
+#'  a variable previously defined in `$PRED` or `$ERROR`.
+#' @param .suffix a suffix for the simulation directory. Will be prefixed by
+#'  the model id of `.mod`.
+#' @inheritParams copy_model_from
+#'
+#' @return S3 object of class `bbi_nmsim_model`.
+#' @export
+new_sim_model <- function(
+    .mod,
+    n = 100,
+    seed = 1234,
+    replicate_var = "REPI",
+    .suffix = "sim",
+    .inherit_tags = TRUE,
+    .overwrite = FALSE
+){
+  check_model_object(.mod, NM_MOD_CLASS)
+  check_model_for_sim(.mod, replicate_var = replicate_var)
+
+  # Create new simulation model object
+  model_dir <- get_model_working_directory(.mod)
+  sim_dir <- glue("{get_model_id(.mod)}-{.suffix}")
+
+  .sim_mod <- copy_model_from(
+    .parent_mod = .mod,
+    .new_model = sim_dir,
+    .add_tags = "SIMULATION",
+    .inherit_tags = .inherit_tags,
+    .update_model_file = TRUE,
+    .overwrite = .overwrite
+  )
+
+  .sim_mod[[YAML_MOD_TYPE]] <- "nmsim"
+  .sim_mod <- save_model_yaml(.sim_mod)
+
+  # Change problem statement
+  prob <- glue("Simulation of model {get_model_id(.mod)}")
+  modify_prob_statement(.sim_mod, prob)
+
+  # Set up simulation
+  setup_sim_run(.sim_mod, n = n, seed = seed, replicate_var = replicate_var)
```

Comment:
Can we save `n` and `seed` in the model object so we can get it later?

Reply:
I was thinking about this. Are you thinking of just adding those elements to the corresponding `yaml`?

Reply:
cc'ing @seth127 in case he has any ideas for this. At the moment was considering adding these directly to the yaml, but maybe tags would be suitable for this (assume not)?